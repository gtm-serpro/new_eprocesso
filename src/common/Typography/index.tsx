import classNames from "classnames";
import { MouseEventHandler } from "react";

import { getMargin } from "@/helpers/styles";
import { BrColor } from "@/types/colors";
import { Spacing } from "@/types/layout";
import { FontSize, FontWeight, TextAlign } from "@/types/text";

type ElementType = "div" | "span" | "p" | "h1" | "h2" | "h3" | "h4" | "h5" | "h6";

export type TypographyProps = {
  children: React.ReactNode;
  className?: string;
  id?: string;
  size?: FontSize;
  align?: TextAlign;
  weight?: FontWeight;
  color?: BrColor;
  margin?: Spacing;
  htmlElement?: ElementType;
  onClick?: MouseEventHandler<HTMLSpanElement>;
};

const Typography = (props: TypographyProps) => {
  const { size, align, weight, color, className, children, margin, htmlElement, ...rest } = props;

  const bgHighlight = color || "default";

  const classes = classNames(className, getMargin(margin), {
    [`text-${size}`]: !!size,
    [`text-${weight}`]: !!weight,
    [`text-${color}`]: !!color,
    [`text-${align}`]: !!align,
    [`ebr-clickable-${bgHighlight}`]: !!rest.onClick,
    "ebr-clickable-text": !!rest.onClick,
  });

  const Wrapper = htmlElement || "div";

  return (
    <Wrapper className={classes} {...rest}>
      {children}
    </Wrapper>
  );
};

export default Typography;
