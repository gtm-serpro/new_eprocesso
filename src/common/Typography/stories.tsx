import { Meta, Story } from "@storybook/react";

import Typography, { TypographyProps } from ".";

export default {
  title: "Layout e Tipografia/Typography",
  component: Typography,
} as Meta;

export const Default: Story<TypographyProps> = (args) => (
  <Typography {...args} onClick={undefined}>
    Conte√∫do Texto
  </Typography>
);

const onClick = () => alert("click");

export const ComEventoClique: Story<TypographyProps> = () => (
  <div>
    <div>
      <Typography onClick={onClick}>Passe o cursor do mouse aqui</Typography>
    </div>
    <div>
      <Typography onClick={onClick} color="primary-default">
        Passe o cursor do mouse aqui
      </Typography>
    </div>
    <div>
      <Typography onClick={onClick} color="secondary-01">
        Passe o cursor do mouse aqui
      </Typography>
    </div>
    <div>
      <Typography onClick={onClick} color="info">
        Passe o cursor do mouse aqui
      </Typography>
    </div>
    <div>
      <Typography onClick={onClick} color="success">
        Passe o cursor do mouse aqui
      </Typography>
    </div>
    <div>
      <Typography onClick={onClick} color="warning">
        Passe o cursor do mouse aqui
      </Typography>
    </div>
    <div>
      <Typography onClick={onClick} color="danger">
        Passe o cursor do mouse aqui
      </Typography>
    </div>
  </div>
);
