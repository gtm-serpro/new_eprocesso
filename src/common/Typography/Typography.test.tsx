import { render } from "@testing-library/react";
import Typography from ".";

describe("<Typography/>", () => {
  it("deveria renderizar e conferir snapshot", () => {
    const { container } = render(
      <Typography weight="bold" size="up-03" color="primary-default" onClick={() => ""}>
        Texto
      </Typography>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("deveria renderizar com o tamanho informado", () => {
    const { getByText } = render(<Typography size="up-03">Texto</Typography>);
    expect(getByText("Texto")).toHaveClass("text-up-03");
  });

  it("deveria renderizar com o weight informado", () => {
    const { getByText } = render(<Typography weight="extra-bold">Texto</Typography>);
    expect(getByText("Texto")).toHaveClass("text-extra-bold");
  });

  it("deveria renderizar com a color informada", () => {
    const { getByText } = render(<Typography color="secondary-01">Texto</Typography>);
    expect(getByText("Texto")).toHaveClass("text-secondary-01");
  });

  it("deveria adicionar hover se clickable", () => {
    const { getByText } = render(<Typography onClick={() => ""}>Texto</Typography>);
    expect(getByText("Texto")).toHaveClass("ebr-clickable-text");
  });
});
