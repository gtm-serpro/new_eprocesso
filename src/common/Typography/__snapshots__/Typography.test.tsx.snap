// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography/> deveria renderizar e conferir snapshot 1`] = `
<div
  class="text-up-03 text-bold text-primary-default ebr-clickable-primary-default ebr-clickable-text"
>
  Texto
</div>
`;
